### ISSUE TITLE
ldap3.add modifies the passed in values

### ISSUE THREAD

**Issue by daenney** [cannatag/ldap3#487]
When you have code like this:

```python
conn = ldap3.Connection(....)
initial = {'attr': [value, value], 'other_attr': [value, value]}
print(initial)
conn.add('dn', [list, of, object, Classes], initial)
print(initial)
```

The expectation would be that the two `print` statements result in identical output. However, when passing my dict of attributes into `add` that dict gets modified by the add operation to include the object classes I passed in as a list (the second argument).

This showed up in some of my test cases where I compiled an initial dict of attributes, did the add and after the add had it log that an entity was created with `initial.keys()`. At that point my tests started to fail b/c  `sorted(initial.keys())` before and after the add operation were no longer the same.

Imho this shouldn't happen. If `ldap3.add` goes about and changes the dict I push into it, it should use `copy.deepcopy()` to create its own since there's no way for ldap3 to know if I might be using that dict somewhere else too, or access it later. I've fixed this in my own code by doing an `add(... copy.deepcopy(inital))` as a workaround.

**Comment by cannatag** [comment]
I agree with you, this was not intentional. Will fix it in the next relase of ldap3.

**Comment by cannatag** [comment]
Hi, I've fixed this issue and added a test case for it. Can you check the code in dev?

**Comment by daenney** [comment]
That works. Thanks for fixing!

**Comment by cannatag** [comment]
Fixed in v2.5.
